derive-eip712/src/lib.rs

```diff
@@ -0,0 +1,399 @@
+                    current_arg = ident.to_string();
+                } else if let quote::__private::TokenTree::Literal(literal) = item {
+                    match current_arg.as_ref() {
+                        "name" => {
+                            domain.name = literal.to_string().replace("\"", "");
+                        }
+                        "version" => {
+                            domain.version = literal.to_string().replace("\"", "");
+                        }
+                        "chain_id" => {
+                            domain.chain_id = literal
+                                .to_string()
+                                .parse::<U256>()
+                                .expect("failed to parse chain id from macro arguments");
+                        }
+                        "verifying_contract" => {
+                            domain.verifying_contract = literal
+                                .to_string()
+                                .replace("\"", "")
+                                .parse::<H160>()
+                                .expect("failed to parse verifying contract");
+                        }
+                        _ => {
+                            panic!("expected arguments: 'name', 'version', 'chain_id' and 'verifying_contract'; found: {}", current_arg);
+                        }
+                    }
+                }
+            }
+        };
+
+        domain
+    }
+}
+
+// Convert rust types to enc types. This is used in determining the type hash;
+// NOTE: this is not an exhaustive list
+fn parse_field_type(field_type: String) -> String {
+    match field_type.as_ref() {
+        "U128" => "uint128",
+        "U256" => "uint256",
+        "H128" => "bytes16",
+        "H160" => "address",
+        "H256" => "bytes32",
+        "String" => "string",
+        "Bytes" => "bytes",
+        "Vec<u8>" => "bytes",
+        "Vec<U128>" => "uint128[]",
+        "Vec<U256>" => "uint256[]",
+        "Vec<H128?" => "bytes16[]",
+        "Vec<H160>" => "address[]",
+        "Vec<H256?" => "bytes32[]",
+        "Vec<String>" => "string[]",
+        "Vec<Bytes>" => "bytes[]",
+        _ => {
+            // NOTE: This will fail if the field type does not match an ethereum type;
+            &field_type
+        }
+    }
+    .to_string()
+}
+
+// Parse the field type from the derived struct
+fn parse_field(field: &syn::Field) -> String {
+    let field_path = match &field.ty {
+        syn::Type::Path(p) => p,
+        _ => {
+            panic!("field type must be a path")
+        }
+    };
+
+    let segment = field_path
+        .path
+        .segments
+        .first()
+        .expect("field must have a type");
+
+    let mut field_type = segment.ident.to_string();
+
+    if let syn::PathArguments::AngleBracketed(arguments) = &segment.arguments {
+        field_type.push('<');
+        for arg in &arguments.args {
```

Comment:
Currently, this works well if there is only one argument within the angle bracket. Should likely throw an error if there are more than one arguments. Trying to think of a use case where multiple arguments (e.g. HashMap<String, String>) would be mapped to an ethereum type.

Reply:
No longer using this method; deprecating in favor of `find_parameter_type`